cli/cli.go

```diff
@@ -112,48 +132,189 @@ func newCli(
 			password := addCmd.StringOpt("p password", "", "Password")
 
 			addCmd.Action = func() {
-				exitWith("", core.Auth().Add(context.TODO(), *resources, *username, *password))
+				node, err := nodeProvider.CreateNodeIfNotExists()
+				if nil != err {
+					exitWith("", err)
+				}
+
+				exitWith(
+					"",
+					node.AddAuth(
+						context.TODO(),
+						model.AddAuthReq{
+							Resources: *resources,
+							Creds: model.Creds{
+								Username: *username,
+								Password: *password,
+							},
+						},
+					),
+				)
 			}
 		})
 	})
 
 	cli.Command("events", "Stream events", func(eventsCmd *mow.Cmd) {
 		eventsCmd.Action = func() {
-			exitWith("", core.Events(context.TODO()))
+			node, err := nodeProvider.CreateNodeIfNotExists()
```

Comment:
Maybe let the dust settle and if we feel it's worth the tradeoffs post these changes we can always add one. This PR gets rid of the NodeHandle interface and makes NodeProvider return a Node directly so that kindof eats away at the value of having another layer of abstraction IMHO. The whole reason we don't new up the node in one place is just because some commands don't require one; it's not really terrible having 3 lines to obtain one only for those that do. Pros/cons of both approaches...